src/nspawn/nspawn-network.c

```diff
@@ -743,3 +778,40 @@ int remove_veth_links(const char *primary, char **pairs) {
 
         return 0;
 }
+
+int network_iface_pair_parse_and_test(const char* iftype, char ***l, const char *p) {
+        _cleanup_free_ char *a = NULL, *b = NULL;
+        int r;
+
+        r = extract_first_word(&p, &a, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
+        if (r < 0)
+                return log_error_errno(r, "Failed to extract first word in %s parameter: %m", iftype);
+        if (r == 0 || !ifname_valid(a))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "%s, interface name not valid: %s", iftype, a);
+
+        r = test_network_interface_initialized(a);
+        if (r < 0)
+                return r;
+
+        r = extract_first_word(&p, &b, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
+        if (r < 0)
+                return log_error_errno(r, "Failed to extract second word in %s parameter: %m", iftype);
+        if (r == 0 || !ifname_valid(b)) {
```

Comment:
this doesn't look right. if the second ifame is not valid this should generate and error, and not just silently be eaten up...

Reply:
I agree with that. But i wanted to be

Reply:
Ok.
The second parameter is optional - empty string means it was not provided by user.